### PR TITLE
7516 Inspection additional user testing with outline shader effect #9

### PR DESCRIPTION
**Pull request by ElderOrb**
**Test plan**

Desired behavior:
(1) far grab behavior:
half trigger -> blue laser -> blue outline on object.
full trigger -> red laser -> red outline on object.
(2) marketplace item:
half trigger ->orange outline to override blue outline. (info) will show up after half second, with orange filing on object. click (info) will open up certificate, which has a button linking to marketplace item page
full trigger -> save as original behavior stated above

https://highfidelity.fogbugz.com/f/cases/7516/Inspection-additional-user-testing-with-outline-shader-effect#BugEvent.47938

### CODE REVIEW

File: scripts/system/controllers/controllerModules/farActionGrabEntity.js [hunk 1]
```diff
@@ -130,30 +134,94 @@ Script.include("/~/system/libraries/controllers.js");
...
+
+        this.updateOutline = function (intersection, laserPointerMode) {
+            var selectedObjectID = intersection.type !== RayPick.INTERSECTED_NONE ? intersection.objectID : null;
+            var selectedObjectType = null;
+
+            if (laserPointerMode === LASER_POINTER_MODE_HOLD) {
+                this.disableOutline();
+            } else {
+                if (selectedObjectID !== this.outlinedObjectID || laserPointerMode !== this.lastLaserPointerMode) {
+                    if (this.entityWithContextOverlay && (selectedObjectID !== this.outlinedObjectID)) // entity has context overlay ? keep prev outline
```
Comment (thoys) [inline scripts/system/controllers/controllerModules/farActionGrabEntity.js]: Please always use brackets even if it is a single If statement (style rule 4.1.11 , https://docs.highfidelity.com/build-guide/coding-standards )

File: scripts/system/controllers/controllerModules/farActionGrabEntity.js [hunk 2]
```diff
@@ -166,6 +234,9 @@ Script.include("/~/system/libraries/controllers.js");
         this.laserPointerOff = function() {
             LaserPointers.disableLaserPointer(this.laserPointer);
             LaserPointers.disableLaserPointer(this.headLaserPointer);
+
+            if(!this.entityWithContextOverlay) // for some reasons laserPointerOff gets also called right after showing context overlay
```
Comment (thoys) [inline scripts/system/controllers/controllerModules/farActionGrabEntity.js]: - Please always use brackets even if it is a single If statement (style rule 4.1.11 , https://docs.highfidelity.com/build-guide/coding-standards )

- And add missing space between `if` and the `(`

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/3297/HighFidelity-Beta-PR11360-7c57c377be78c80091f1f6e29c159cd7126f9a5b.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/3297/HighFidelity-Beta-PR11360-7c57c377be78c80091f1f6e29c159cd7126f9a5b.exe

**Comment by ElderOrb** [comment]
Will do, thanks!

**Comment by ElderOrb** [comment]
is it supported by VS 2017 Community edition?

**Comment by ElderOrb** [comment]
While configuring lint with VS 2017 I've noticed lint config is a bit broken. https://github.com/highfidelity/hifi/pull/11363

File: .eslintrc.js [hunk 3]
```diff
@@ -57,7 +57,9 @@ module.exports = {
         "print": false,
         "RayPick": false,
         "LaserPointers": false,
-        "ContextOverlay": false
+        "ContextOverlay": false,
+	"Selection" : false,
+	"Render" : false,
```
Comment (thoys) [inline .eslintrc.js]: Thanks for updating, but please convert these tabs to spaces :)

Comment (ElderOrb) [reply .eslintrc.js]: omg... :) ok.

Comment (ElderOrb) [reply .eslintrc.js]: done

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/3307/HighFidelity-Beta-PR11360-396fd68ea58936218a92df72468a6407e62c7a33.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/3307/HighFidelity-Beta-PR11360-396fd68ea58936218a92df72468a6407e62c7a33.exe

File: .eslintrc.js [hunk 4]
```diff
@@ -58,6 +58,8 @@ module.exports = {
         "RayPick": false,
         "LaserPointers": false,
         "ContextOverlay": false,
+        "Selection" : false,
+        "Render" : false,
```
Comment (thoys) [inline .eslintrc.js]: And please omit the space on the left of the colon for consistency :)

Comment (ElderOrb) [reply .eslintrc.js]: sure :)

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/3318/HighFidelity-Beta-PR11360-3f51f68bd7acfbba61896e3ed4ffb4188f7bbe4a.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/3318/HighFidelity-Beta-PR11360-3f51f68bd7acfbba61896e3ed4ffb4188f7bbe4a.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/3342/HighFidelity-Beta-PR11360-2018bbc9c9a58ea7a4e840e108848dc4b0b01123.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/3342/HighFidelity-Beta-PR11360-2018bbc9c9a58ea7a4e840e108848dc4b0b01123.exe

**Comment by changxc** [comment]
we can close this branch
Inspection feature has been merged into master by discovery / commerce team

**Pull request closed without merging**